accepted/future-releases/static-extension-methods/feature-specification.md

```diff
@@ -46,214 +46,205 @@ extension MyFancyList<T> on List<T> {
+If an extension conflicts with, or is shadowed by, another declaration, and you need to access it by name anyway, it can be imported with a prefix and the name referenced through that prefix.
 
-Example:
+*Rationale*: We want users to have control over which extensions are available. They control this through the imports and declarations used to include declarations into the import scope or declaration scope of the library. The typical ways to control the import scope is using `show` /`hide` in the imports or importing into a prefix scope. These features work exactly the same for extensions. On the other hand, we do not want extension writers to have to worry too much about name clashes for their extension names since most extension members are not accessed through their name anyway. In particular we do not want them to name-mangle their extensions in order to avoid hypothetical conflicts. So, all imported extensions are considered in scope, and choosing between the individual extensions is handled as described in the next section. You only run into problems with the extension name if you try to use the name. That way you can import two extensions with the same name and use the members without issue (as long as they don't conflict in an unresolvable way), even if you can only refer to *at most* one of them by name.
 
-```dart
-extension TypedEquals<T> on T {
-  bool equals(T value) => this == value;
-}
-```
+You still cannot *export* two extensions with the same name. The rules for export makes it a compile-time error to add two declarations with the same name to the export scope of a library.
 
-Using such an extension as:
+##### Applicability
 
-```dart
-Object o = ...;
-String s = ...;
-print(s.equals(o));  // Compile-time type error.
-```
+An extension `E` is *applicable* to a simple or composite member invocation with corresponding member *basename* *m* and target expression `e`, where `e` has static type *S*, if
+
+- The invocation is an *instance* member invocation. That is the case if the expression `e` does not denote a prefix or a class, mixin or extension declaration *(then the member invocation would be a static invocation)*. An instance member invocation on `e` will always begin by evaluating `e` to an object, and then continue by performing an instance member invocation on that object.
+- The type *S*  does not have a member with the basename  *m*. For this, the types `dynamic`and `Never` are considered as having all member names, and an expression of type `void` cannot occur as the target of a member invocation, so none of these can ever have applicable extensions. Function types and the type `Function` are considered as having a `call` member. *This ensure that if there is an applicable extension, the existing invocation would otherwise be a compile-time error*. Members of `Object` exists on all types, so they can never be the target of implicit member invocations (they can also not be declared as extension members).
```

Comment:
The NNBD feature spec makes member accesses on `Never` an error, so it should not be necessary to say that it has all members.

Reply:
Moved it to be with `void` instead.